iped-app/src/main/java/iped/app/home/config/ConfigPanel.java

```diff
@@ -0,0 +1,424 @@
+            textFieldTskJarPath.setText( (pluginConfig.getTskJarFile() != null )? pluginConfig.getTskJarFile().getAbsolutePath() : "" );
+        }
+        textFieldPluginFolder.setText( (pluginConfig.getRelativePluginFolder() != null )? pluginConfig.getRelativePluginFolder() : "" );
+    }
+
+    /**
+     * Save all changes made by user
+     */
+    private void saveConfiguration(){
+        try {
+            //Save LocaleConfig
+            LocaleConfig localeConfig = ConfigurationManager.get().findObject(LocaleConfig.class);
+            Languages selectedLanguage = ((Languages) comboBoxLocale.getSelectedItem());
+            if( ! localeConfig.getLocale().toLanguageTag().equals( selectedLanguage.getLanguageTag() ) ){
+                JOptionPane.showMessageDialog(this, Messages.get("Home.LanguageChangeMessage"), Messages.get("Home.LanguageChangeTitle"), JOptionPane.WARNING_MESSAGE);
+            }
+            localeConfig.setLocale( selectedLanguage.getLanguageTag() );
+            localeConfig.getPropertie().saveOnFile(CasePathManager.getInstance().getLocalConfigFile());
+
+
+            //Force LocalConfig to reload possible changes made by LocaleConfig
+            ConfigurationManager.get().reloadConfigurable(LocalConfig.class);
+            LocalConfig localConfig = ConfigurationManager.get().findObject(LocalConfig.class);
+            //Set changes on LocalConfig Configurable class
+            localConfig.setIndexTempOnSSD( checkBoxIndexTempOnSSD.isSelected() );
+            localConfig.getPropertie().setProperty(LocalConfig.INDEX_TEMP, textFieldIndexTemp.getText());
+            localConfig.setNumThreads((Integer) spinnerThreads.getValue());
+            localConfig.getPropertie().setProperty(LocalConfig.HASH_DB, textFieldHashesDB.getText());
+            //Save LocalConfig modifications to file
+            localConfig.getPropertie().saveOnFile(CasePathManager.getInstance().getLocalConfigFile());
+
+            //Force PluginConfig to reload possible changes made by LocalConfig
+            ConfigurationManager.get().reloadConfigurable(PluginConfig.class);
+            PluginConfig pluginConfig = ConfigurationManager.get().findObject(PluginConfig.class);
+            //Set changes on PluginConfig Configurable class
+            pluginConfig.setPluginFolder(textFieldPluginFolder.getText());
+            if (!RUNNIG_ON_WINDOWS) {
+                if ((textFieldTskJarPath.getText() == null) || (!textFieldTskJarPath.getText().trim().isEmpty()))
+                    pluginConfig.setTskJarFile( textFieldTskJarPath.getText() != null ? textFieldTskJarPath.getText().trim() : null );
+            }
+            pluginConfig.getPropertie().saveOnFile(CasePathManager.getInstance().getLocalConfigFile());
+
+            //Force LocalConfig to reload changes made by PluginConfig
+            ConfigurationManager.get().reloadConfigurable(LocalConfig.class);
+            ConfigurationManager.get().reloadConfigurable(PluginConfig.class);
```

Comment:
@thiagofuer, these reloadings are preventing the update of LocalConfig and PluginConfig components' state, so that when the user saves a configuration and goes back to the ConfigPanel, the values are reset (null)

Reply:
also the checkBoxEnableDisableHashDb is not working as expected for me, can you confirm this?

Reply:
Hi sorry for the delay. 
I'll check the considerations you gave me

Reply:
if there is another way to solve this problem please help me to find it..